lib/pdk/util/ruby_version.rb

```diff
@@ -48,12 +48,19 @@ def scan_for_packaged_rubies
         end
 
         def default_ruby_version
-          # For now, the packaged versions will be using default of 2.4.5
-          # FIXME: make this dynamic
-          return '2.4.5' if PDK::Util.package_install?
-
-          # TODO: may not be a safe assumption that highest available version should be default
-          latest_ruby_version
+          @default_ruby_version ||= if PDK::Util.package_install?
+                                      # Default to the ruby that supports the latest puppet gem. If you wish to default to a
+                                      # specific Puppet Gem version use the following example;
+                                      #
+                                      # PDK::Util::PuppetVersion.find_gem_for('5.5.10')[:ruby_version]
+                                      #
+                                      PDK::Util::PuppetVersion.latest_available[:ruby_version]
+                                    else
+                                      # TODO: may not be a safe assumption that highest available version should be default
+                                      # WARNING Do NOT use PDK::Util::PuppetVersion.*** methods as it can recurse into this
```

Comment:
This conflicts with the code in the true side of this if statement where `PDK::Util::PuppetVersion.latest_available` is used.

Reply:
Which bit conflicts?  The warning?

Reply:
To clarify, when not in a packaged install there exists this callpath which can cause the recursion loop

PDK::Util::PuppetVersion.find_gem ->
PDK::Util::PuppetVersion.find_in_rubygems ->
PDK::Util::RubyVersion.default_ruby_version

Packaged installs use a deterministic code path using the on disk files so the recursion doesn't happen.

Reply:
Because this behaviour may change in the future I've left the comment as a warning for future-us